File: MainApp/src/main/java/com/sleticalboy/dailywork/ui/activity/TransitionUI.java
package com.sleticalboy.dailywork.ui.activity;

import android.content.Intent;
import android.view.View;

import com.sleticalboy.dailywork.R;
import com.sleticalboy.dailywork.base.BaseActivity;

import androidx.core.app.ActivityOptionsCompat;

/**
 * Created by AS 3.3 on 19-1-6.
 *
 * @author leebin
 */
public class TransitionUI extends BaseActivity {

    @Override
    protected int layoutResId() {
        return R.layout.transition_activity;
    }

    @Override
    protected void initView() {
        final View rootView = findViewById(R.id.rootView);
        final View fakeSearch = findViewById(R.id.fakeSearch);
        final View realSearch = findViewById(R.id.realSearch);
        if (getIntent().getStringExtra("place_holder") == null) {
            fakeSearch.setOnClickListener(this::showShareAnimation);
            fakeSearch.setVisibility(View.VISIBLE);
            realSearch.setVisibility(View.INVISIBLE);
            rootView.setAlpha(1.0f);
        } else {
            realSearch.setVisibility(View.VISIBLE);
            fakeSearch.setVisibility(View.GONE);
            rootView.setAlpha(0.75f);
        }
    }

    private void showShareAnimation(View view) {
        final Intent intent = new Intent(this, TransitionUI.class);
        // final Rect rect = new Rect();
        // view.getGlobalVisibleRect(rect);
        // intent.setSourceBounds(rect);
        intent.putExtra("place_holder", "place_holder");
        final ActivityOptionsCompat options = ActivityOptionsCompat.makeSceneTransitionAnimation(this, view, "share_element_test");
        startActivity(intent, options.toBundle());
        overridePendingTransition(0, 0);
    }

    private void jump(View sharedElement) {
        getSupportFragmentManager()
                .beginTransaction()
                .addSharedElement(sharedElement, "shared_element_test")
                .replace(R.id.container, null)
                .addToBackStack(null)
                .commit();
    }
}

File: MainApp/src/main/java/com/sleticalboy/learning/anims/TransitionUI.kt
package com.sleticalboy.learning.anims

import android.content.Intent
import android.view.View
import androidx.core.app.ActivityOptionsCompat
import com.sleticalboy.learning.R
import com.sleticalboy.learning.base.BaseActivity


/**
 * Created by AS 3.3 on 19-1-6.
 *
 * @author leebin
 */
class TransitionUI : BaseActivity() {
    override fun layoutResId(): Int {
        return R.layout.activity_transition
    }

    override fun initView() {
        val rootView = findViewById<View>(R.id.rootView)
        val fakeSearch = findViewById<View>(R.id.fakeSearch)
        val realSearch = findViewById<View>(R.id.realSearch)
        if (intent.getStringExtra("place_holder") == null) {
            fakeSearch.setOnClickListener { view: View -> showShareAnimation(view) }
            fakeSearch.visibility = View.VISIBLE
            realSearch.visibility = View.INVISIBLE
            rootView.alpha = 1.0f
        } else {
            realSearch.visibility = View.VISIBLE
            fakeSearch.visibility = View.GONE
            rootView.alpha = 0.75f
        }
    }

    private fun showShareAnimation(view: View) {
        val intent = Intent(this, TransitionUI::class.java)
        // final Rect rect = new Rect();
        // view.getGlobalVisibleRect(rect);
        // intent.setSourceBounds(rect);
        intent.putExtra("place_holder", "place_holder")
        val options = ActivityOptionsCompat.makeSceneTransitionAnimation(this, view, "share_element_test")
        startActivity(intent, options.toBundle())
        overridePendingTransition(0, 0)
    }

    private fun jump(sharedElement: View) {
        supportFragmentManager
                .beginTransaction()
                .addSharedElement(sharedElement, "shared_element_test")
                .replace(R.id.container, null)
                .addToBackStack(null)
                .commit()
    }
}